docs/shared/recipes/generators/create-sync-generator.md

```diff
@@ -0,0 +1,136 @@
+# Create a Sync Generator
+
+Sync generators are generators that are used to ensure that your file system is in the correct state before a task is run or the CI process is started. From a technical perspective, a sync generator is no different from any other generator, but it has some additional performance considerations and needs to be registered in a particular way.
+
+## Create a Generator
+
+Create a sync generator the same way you would [create any generator](/extending-nx/recipes/local-generators):
+
+```shell
+nx g my-sync-generator --directory=tools/my-plugin/src/generators/my-sync-generator
+```
+
+A sync generator should be able to run without any required options, so update the schema accordingly:
+
+```jsonc {% fileName="tools/my-plugin/src/generators/my-sync-generator/schema.json" %}
+{
+  "$schema": "https://json-schema.org/schema",
+  "$id": "MySyncGenerator",
+  "title": "",
+  "type": "object",
+  "properties": {},
+  "required": []
+}
+```
+
+Sync generators can optionally return an `outOfSyncMessage` to display to users when the sync generator needs to be run.
+
+```ts {% fileName="tools/my-plugin/src/generators/my-sync-generator/my-sync-generator.ts" %}
+import { Tree } from '@nx/devkit';
+import { SyncGeneratorResult } from 'nx/src/utils/sync-generators';
+
+export async function mySyncGenerator(
```

Comment:
This one and the one below seem like examples of global sync generators. Maybe worth calling them `myGlobalSyncGenerator` and adding a more relevant example of a task one?